packages/utils/src/address.ts

```diff
@@ -33,3 +33,56 @@ export const isValidPrivateKey = (privateKey: string) =>
 export { isValidAddress, isValidChecksumAddress, toChecksumAddress } from 'ethereumjs-util'
 
 export const NULL_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+export const findAddressIndex = (address: Address, addresses: Address[]) =>
+  addresses.findIndex((x) => eqAddress(x, address))
+
+export const mapAddressListOnto = (oldAddress: Address[], newAddress: Address[]) => {
+  const oldAddressIndex: Array<{
+    address: Address
+    index: number
+  }> = oldAddress.map((x: Address, index: number) => ({ address: normalizeAddress(x), index }))
+
+  const newAddressIndex: Array<{
+    address: Address
+    index: number
+  }> = newAddress.map((x: Address, index: number) => ({ address: normalizeAddress(x), index }))
+
+  oldAddressIndex.sort((a, b) => a.address.localeCompare(b.address))
+  newAddressIndex.sort((a, b) => a.address.localeCompare(b.address))
+  const res = [...Array(oldAddress.length).fill(-1)]
+
+  for (let i = 0, j = 0; i < oldAddress.length && j < newAddress.length; ) {
+    const cmp = oldAddressIndex[i].address.localeCompare(newAddressIndex[j].address)
+    if (cmp < 0) {
+      i++
+    } else if (cmp > 0) {
+      j++
+    } else {
+      // Address is present in both lists
+      res[oldAddressIndex[i].index] = newAddressIndex[j].index
+      i++
+      j++
+    }
+  }
+  return res
+}
+
+export function mapAddressListDataOnto<T>(
```

Comment:
Where is this used? Not seeing it elsewhere in the PR. And is it something we'll need often enough to warrant putting it in the shared utils package?

Reply:
This is only used by a slasher bot in a different PR that we're currently choosing not to publish: https://github.com/celo-org/celo-monorepo/compare/jfoutts/slasher-bot#diff-d81498e3cf752be28e3ca22b0b6ab510R84 But it presumably would be wanted by other developers of slasher bots.